editor/language_server/src/lsp.rs

```diff
@@ -195,6 +201,20 @@ impl LanguageServer for SlideLS {
         self.close(&uri);
     }
 
+    async fn completion(&self, params: CompletionParams) -> Result<Option<CompletionResponse>> {
+        let TextDocumentPositionParams {
+            text_document: TextDocumentIdentifier { uri },
+            position,
+        } = params.text_document_position;
+        self.client
+            .log_message(MessageType::Info, format!("Got completion request"))
+            .await;
+        let program_info = self.get_program_info(&uri);
+
+        let completions = services::get_completions(position, program_info.deref());
+        Ok(completions)
+    }
```

Comment:
also you return result, but theres no case where it isn't ok.

Reply:
its required by the trait